From: srinivas pulukuru <srinivas.pulukuru@ti.com>
Date: Fri, 13 May 2011 16:36:56 +0000 (-0500)
Subject: OMAP4 DSS HDMI: Fix release of L3 constraints twice
X-Git-Url: http://omapzoom.org/?p=kernel%2Fomap.git;a=commitdiff_plain;h=67de076211183f0ecdf1ddf034176dd2b45173f9

OMAP4 DSS HDMI: Fix release of L3 constraints twice

L3 contraints were getting released twice due a incorrect check
Constraints are release and set along with 48MHz Phy clock

Change-Id: If1361d07b6c0076e758e543ac56c7084e2319b48
Signed-off-by: Srinivas Pulukuru <srinivas.pulukuru@ti.com>
---

diff --git a/drivers/video/omap2/dss/hdmi.c b/drivers/video/omap2/dss/hdmi.c
index 52a745b..3946ced91dee9cdca62578443eb545cbd19b9b0c 100644
--- a/drivers/video/omap2/dss/hdmi.c
+++ b/drivers/video/omap2/dss/hdmi.c
@@ -332,25 +332,25 @@ static int hdmi_set_48Mhz_l3_cstr(struct omap_dss_device *dssdev, bool enable)
 			hdmi_opt_clk_state = enable ? true : false;
 		else
 			goto err;
-	}
 
 #ifdef CONFIG_OMAP_PM
-	DSSINFO("%s L3-200Mhz constraint\n\n",
-		enable ? "Set" : "Release");
-	r = omap_pm_set_min_bus_tput(
-		&dssdev->dev,
-		OCP_INITIATOR_AGENT,
-		enable ? 200 * 1000 * 4 : -1);
-	if (r)
-		DSSERR("unable to %s L3-200Mhz constraint\n",
-		       enable ? "release" : "set");
-
-	r = omap_pm_set_max_sdma_lat(&pm_qos_handle,
-				       enable ? 10 : -1);
-	if (r)
-		DSSERR("Unable to %s core cstr\n",
-		       enable ? "set" : "remove");
+		DSSINFO("%s L3-200Mhz constraint\n\n",
+			enable ? "Set" : "Release");
+		r = omap_pm_set_min_bus_tput(
+			&dssdev->dev,
+			OCP_INITIATOR_AGENT,
+			enable ? 200 * 1000 * 4 : -1);
+		if (r)
+			DSSERR("unable to %s L3-200Mhz constraint\n",
+			       enable ? "release" : "set");
+
+		r = omap_pm_set_max_sdma_lat(&pm_qos_handle,
+					       enable ? 10 : -1);
+		if (r)
+			DSSERR("Unable to %s core cstr\n",
+			       enable ? "set" : "remove");
 #endif
+	}
 
 err:
 	return ret;